ipatests/test_integration/test_upgrade.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Module provides tests to verify that the upgrade script works.
+"""
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_plugins.integration import tasks
+from ipaserver.install import sysupgrade
+
+OLD_VERSION = '1.0.0-1.fc9'
+
+
+class TestUpgrade(IntegrationTest):
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=False)
+
+    def test_invoke_upgrader(self):
+        # For a new version in to require the upgrade to run
+        sysupgrade.set_upgrade_state('ipa', 'data_version',
```

Comment:
Unless I'm mistaken, this command would be run on the controller from where the tests are launched (which can be different from the master) and will trigger an error.
Moreover, I don't think you need to tweak the data_version for ipa-server-upgrade to run, the upgrade can be executed even without changing the version.

Reply:
Right, of course. My purpose was to be absolutely sure that the upgrade was successful. I can remove this and revisit in the future if need be.